### PR TITLE
use same search box on phobos pages as elsewhere

### PR DESCRIPTION
**Pull request by aG0aep6G**
Also DRY out other minor differences between doc.ddoc and std.ddoc.

### CODE REVIEW

File: std.ddoc [hunk 1]
```diff
@@ -1,55 +1,13 @@
...
-            <img id='mobile-logo' width="50" height="38" alt="D Logo" src="http://dlang.org/images/dlogo.png">
-            $(DIVID mobile-d-language,
-                <a href="http://dlang.org">D$(NBSP)Programming Language</a>
-            )
-        )
-    )
-)
-$(DIVID navigation, $(TOP)
-$(NAVIGATION_PHOBOS)
+    $(TOP_SEARCHBOX ,selected,)
```
Comment (aG0aep6G) [inline std.ddoc]: This isn't particularly beautiful, but I couldn't think of a nicer way to pre-select an &lt;option&gt;.

Comment (CyberShadow) [inline std.ddoc]: How about:
```
Macros:
    SEARCH_DEFAULT_STD=selected
```

File: doc.ddoc [hunk 2]
```diff
@@ -67,6 +19,7 @@ $(SCRIPTLOAD js/cssmenu.js)
 </html>
 _=
 
+ROOT = .
```
Comment (CyberShadow) [inline doc.ddoc]: Why not empty string here, and use `$(ROOT)js/...` ?

File: doc.ddoc [hunk 3]
```diff
@@ -1,60 +1,12 @@
...
-<link rel="stylesheet" href="css/print.css" media="print" />
-<link rel="stylesheet" href="css/cssmenu.css">
-<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css">
-<link rel="shortcut icon" href="favicon.ico" />
-<meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=0.1, maximum-scale=10.0" />
-</head>
+$(DOCTYPE_OPENHTML_COPYRIGHT_HEAD)
 <body id='$(TITLE)'>
 $(SCRIPT document.body.className += ' have-javascript';)
 $(DIVID top,
```
Comment (CyberShadow) [inline doc.ddoc]: Why have two macros when you can have one? This bit is almost identical too.

File: std.ddoc [hunk 4]
```diff
@@ -68,18 +26,7 @@ $(SCRIPT jQuery(document).ready(listanchors);)
 </html>
```
Comment (CyberShadow) [inline std.ddoc]: More DRY potential here

File: doc.ddoc (hunk 2 above)
Comment (aG0aep6G) [inline doc.ddoc]: I think

    ROOT = .
    ROOT = ..
    $(ROOT)/foo

are clearer and easier to get right than

    ROOT =
    ROOT = ../
    $(ROOT)foo

But if there's an upside to doing it the other way, I can do that of course.

**Comment by CyberShadow** [comment]
Thanks, I was going to do this at some point as well. But I think more refactoring can be done :)

You can move stuff out of the inline HTML code by specifying it as a macro with different value in `std.ddoc` and `doc.ddoc`, then using it in `dlang.org.ddoc`.

File: doc.ddoc (hunk 2 above)
Comment (CyberShadow) [inline doc.ddoc]: The upside is that HTTP requests won't have a `./` prefix.

Comment (aG0aep6G) [inline doc.ddoc]: Hm, shouldn't the browser usually swallow them?

Comment (CyberShadow) [inline doc.ddoc]: Maybe?

Anyway, this is just a nitpick. But a prefix with a trailing slash is a common pattern, e.g. the `--prefix` parameter to `./configure` scripts.

**Comment by aG0aep6G** [comment]
Pushed a commit that consolidates the multiple macros into one TOP.

**Comment by aG0aep6G** [comment]
and one to DRY the scripts

**Comment by CyberShadow** [comment]
OK, not exactly what I had in mind but this LGTM as is. I'll have a go at my own suggestions myself

File: doc.ddoc (hunk 2 above)
Comment (CyberShadow) [inline doc.ddoc]: The current approach has the advantage that `<a href="$(ROOT)">Index</a>` works properly, which allows linking to the main page without hard-coding the `http://dlang.org/` URL.

**Pull request merged**